app/models/ext_management_system.rb

```diff
@@ -260,6 +264,78 @@ def default_endpoint
     default || endpoints.build(:role => "default")
   end
 
+  # TODO: takes multiple connection data
+  # endpoints, authentications, and works out what's what.
+  def connections=(options)
+    options.each do |option|
+      endpoint = endpoint_builder(option[:endpoint])
+      authentication_builder(option[:authentication])
+        # connection_by_role=(option)
+    end
+  end
+
+  def connections
+    roles = endpoints.map(&:role)
+    options = {}
+
+    roles.each do |role|
+      conn = connection_by_role(role)
+      options[role] = conn
+    end
+
+    connections = OpenStruct.new(options)
```

Comment:
Minor, but prefer MiqHashStruct over OpenStruct...probably won't make much difference in this case though.  (on that note, @jrafanie, we should really see if the old reasons for not using OpenStruct still apply to Ruby 2.2+)